Feature/near by facility

Jpuntul opened the pull request:
### 🛠 Proposed Changes:
Create a POI selection page with a reusable POI box widget and navigation to the POI map.

### 📝 Details:

- Implement a structured POI selection layout.
- Create a reusable POI box widget for consistency.
- Enable navigation to the POI map after selection. (restroom only in this update)
- The direction feature is a placeholder for now

### 🔗 Related Issue(s):

- Resolves #58 
- Resolves #65 
- Resolves #66


https://github.com/user-attachments/assets/cf9e7a9c-baea-4bc8-8e71-5cea819e42c3

MeSumo commented:
The previous commit resolved 7 RenderFlex overflow exceptions that were generated while testing the navigation between the main menu and the POI page. This was due to the PoiBoxes having a fixed row too large for certain device sizes. 
<img width="395" alt="image" src="https://github.com/user-attachments/assets/c08190d0-5046-435c-92e3-3755aecdd85c" />

This was resolved by surrounding the Text field in the row with Flexible, such that the text would wrap when it is too long. 
<img width="347" alt="image" src="https://github.com/user-attachments/assets/343b96a6-b91e-4d8e-a2d8-89a5d30403a2" />

MeSumo reviewed (dismissed):
### Tested with a physical Android device (v15)
Selecting the Nearby Facilities option brings us to the POI choice page where we get a list of POIs. Choosing the Restrooms one brings to the POI map view which contains only three text fields for now. The back buttons on both pages bring you to the previous page successfully.

![Screenshot_2025-02-04-12-58-56-91_d7842e130cbf5ce865fe75d1c90bc1e0](https://github.com/user-attachments/assets/0df5fcf2-3222-4493-a556-c28314c0de62)
![Screenshot_2025-02-04-12-59-13-75_d7842e130cbf5ce865fe75d1c90bc1e0](https://github.com/user-attachments/assets/d8556633-3c81-4f79-a9f0-1748af0fba0e)

Niravanaa reviewed (approved):
![image](https://github.com/user-attachments/assets/28119c52-f18f-4e7c-a831-d93f5e2b7a96)
![image](https://github.com/user-attachments/assets/aff1f610-934f-41b6-b894-a4ce8e014c79)

LGTM! Tested on simulated Android.

Jpuntul commented:
I just noticed that the view page was misplaced in the widget folder, so I moved it to the UI folder. I’ve changed all the file paths, so it shouldn't affect anything. Just a quick check, please. @MeSumo @Niravanaa

MeSumo reviewed (approved):
Does in fact still work! Thank you @Jpuntul for the file restructuring! 

Extra thing I noticed, the PoiBoxes compress when the device's keyboard is displayed, such that they are all still in view. Texts and icons are still visible! 
![Screenshot_2025-02-04-17-30-32-29_d7842e130cbf5ce865fe75d1c90bc1e0](https://github.com/user-attachments/assets/10d48ba5-7bdd-40e0-8d22-ea4511099ff6)
